packages/math/stable.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  bnum,
+  OldBigNumber,
+  StableMaths,
+  StablePool,
+  ZERO,
+} from "@balancer-labs/sor";
+import { BigNumber, formatFixed } from "@ethersproject/bignumber";
+import { WeiPerEther as EONE } from "@ethersproject/constants";
+
+type StablePoolPairData = ReturnType<
+  typeof StablePool.prototype.parsePoolPairData
+>;
+const ONE = bnum(1);
+
+function _poolDerivatives(
+  A: BigNumber,
+  balances: OldBigNumber[],
+  tokenIndexIn: number,
+  tokenIndexOut: number,
+  is_first_derivative: boolean,
+  wrt_out: boolean
+): OldBigNumber {
+  const totalCoins = balances.length;
+  const D = StableMaths._invariant(A, balances);
+  let S = ZERO;
+  for (let i = 0; i < totalCoins; i++) {
+    if (i != tokenIndexIn && i != tokenIndexOut) {
+      S = S.plus(balances[i]);
+    }
+  }
+  const x = balances[tokenIndexIn];
+  const y = balances[tokenIndexOut];
+  // A is passed as an ethers bignumber while maths uses bignumber.js
```

Comment:
Couldn't we require this to happen before an expect a bignumber.js object here?

Reply:
This function was copied from @balancer/sor package, since was not exported